### ISSUE TITLE
[docs][guides] Docs page for REPL

### ISSUE THREAD

**Issue by kkemple** [gatsbyjs/gatsby#7484]
A new `repl` command was added to Gatsby CLI but it is currently undocumented.

**Comment by V1shvesh** [comment]
Hey @kkemple!
Can I be assigned to this issue?

**Comment by marcysutton** [comment]
@V1shvesh sorry for the delay, yes we would love for you to take this one! I've added a stub for the REPL and some stuff in the CLI docs in a new PR, once it is merged you can start there: https://github.com/gatsbyjs/gatsby/pull/12132

**Comment by V1shvesh** [comment]
Awesome! I believe it's merged now.

**Comment by V1shvesh** [comment]
Hey @kkemple. Feeling a lil embarassed to admit that I don't have a clue where to start.

I found the [list of REPL commands](https://www.gatsbyjs.org/docs/gatsby-cli/#repl), but I am unable to understand the motivation for such a feature. Please help.

**Comment by pieh** [comment]
@V1shvesh Original idea came from https://github.com/gatsbyjs/gatsby/issues/6952 and it was implemented in https://github.com/gatsbyjs/gatsby/issues/7484 . For the most part it is to allow examining internal state of Gatsby, without resorting to bunch of `console.log` or debug breakpoints while Gatsby is doing it's processing.

It might be useful during development/debugging of plugins, but it does require knowledge of Gatsby internals. Or using it can help people understand internals better (depending how you look at it).

**Comment by giacomocerquone** [comment]
This is awesome, if someone of you needs help tell me, I'm available to work on the doc!

**Comment by marcysutton** [comment]
@gatsbyjs/core is there anything we need to do to the Gatsby REPL to get it into shape for proper documentation? It is in the [Gatsby CLI doc](https://www.gatsbyjs.org/docs/gatsby-cli/#repl), but this issue is about filling in the [top-level docs page](https://www.gatsbyjs.org/docs/gatsby-repl/).

**Comment by n8finch** [comment]
@marcysutton I'd love to give this a go! It might take me a week or two to get familiar with the commands and such to get some basic documentation together, but then we could go from there.

It seems like I can just break up the commands into different pages, and show examples of how and why you would use each, as well as how this could be combined with the GraphQL explorer to be useful to folks. 

I can start with three or four commands and go from there. Thoughts?

**Comment by marcysutton** [comment]
@n8finch sounds awesome! I think one page would be ideal, though. The work slated in #6896 would help to organize it–in the meantime we're using hard-coded tables of contents like on this doc: https://github.com/gatsbyjs/gatsby/blob/master/docs/docs/gatsby-image.md#in-this-doc

**Comment by n8finch** [comment]
@marcysutton perfect, I'll get to work on this! If the issue is quiet for a bit, don't worry, I'm still working... 🤓
